### DISCUSSION TITLE
Led and Power

### DISCUSSION THREAD

**Discussion by ADB-ITA** [CrealityOfficial/Creality_Sonic_Pad#21] [Ideas]
Why is there no "Led" and "power" option on the Sonic Pad home page? In the web interface these options are present, for example the Led option when printing would be useful without going through the macros. 
Thank you.

**Comment by Escrich** [comment]
You can do it with a simple macro

**Reply by ADB-ITA** [reply]
Can you give an example? Or alternatively can you give me a web link?

**Reply by Escrich** [reply]
Just writing fast, may have some errors

```
[output_pin my_pin] # Change my_pin by the name you want to use to refer your pin for example, motor, light, or whatever you want
pin: PG3 # Change PG3 by the pin specification of the pin in your board you want to use, this value is mandatory
value: 0 # This is the value when you start klipper, this value is optional
shutdown_value: 0 # This is the value that your pin will assume if your mcu, (Raspberry or so), goes down, this value is optional

Usage:
you can activate it simply using:

SET_PIN PIN=my_pin VALUE= 1

or deactivate it simply using:

SET_PIN PIN=my_pin VALUE= 0

For example if you write SET_PIN PIN=my_pin VALUE= 1 inside your start_print macro it will start

and then having this line in your end_print macro SET_PIN PIN=my_pin VALUE= 0 it will stop

Then you'll have your, for example light, on, during the whole printing time

```
More info here:
https://www.klipper3d.org/Config_Reference.html#output_pin

Here you'll find my last printer Klipper cofiguration, feel fre to copy whatever you want if you need something:
https://github.com/Escrich/00-Phoenix-SonicPad-00
